spec/pagseguro/authorization_request/request_serializer_spec.rb

```diff
@@ -1,23 +1,285 @@
+        type: phone.at_css('> type').text,
+        area_code: phone.at_css('> areaCode').text,
+        number: phone.at_css('> number').text
+      }
+    end
+
+    expect( phones_parsed.size ).to eq 2
+    expect( phones_parsed ).to eq phones
+  end
+end
 
 describe PagSeguro::AuthorizationRequest::RequestSerializer do
-  let(:credentials) { PagSeguro::ApplicationCredentials.new('app11', 'sada') }
+  let(:credentials) { PagSeguro::ApplicationCredentials.new('app123', 'key123') }
   let(:options) do
     {
       credentials: credentials,
-      permissions: [:checkouts, :notifications],
-      notification_url: 'foo.com',
-      redirect_url: 'bar.com',
-      reference: 'ref1234'
+      permissions: [:checkouts, :searches, :notifications],
+      notification_url: 'http://seusite.com.br/redirect',
+      redirect_url: 'http://seusite.com.br/notification'
     }
   end
-  let(:authorization) { PagSeguro::AuthorizationRequest.new(options) }
-  let(:serializer) { described_class.new(authorization) }
-  let(:params) { serializer.to_params }
 
-  it{ expect(params).to include(credentials: credentials) }
-  it{ expect(params).to include(permissions: 'CREATE_CHECKOUTS,RECEIVE_TRANSACTION_NOTIFICATIONS') }
-  it{ expect(params).to include(notificationURL: 'foo.com') }
-  it{ expect(params).to include(redirectURL: 'bar.com') }
-  it{ expect(params).to include(reference: 'ref1234') }
+  let(:authorization_request) { PagSeguro::AuthorizationRequest.new(options) }
+  let(:serializer) { described_class.new(authorization_request) }
+  let(:xml) { serializer.build_xml }
+  let(:xml_parsed) { Nokogiri::XML(xml) }
+  let(:node) { xml_parsed.at_css('authorizationRequest') }
+
+  it 'encoding should be equal to application encoding' do
+    encoding = xml_parsed.encoding
+    expect(encoding).to eq PagSeguro.encoding
+  end
+
+  it 'must have a authorizationRequest' do
+    expect(node).to be_a_instance_of Nokogiri::XML::Element
+  end
+
+  it 'must have the correct permissions' do
+    permissions = options[:permissions].map{|value| PagSeguro::AuthorizationRequest::PERMISSIONS[value] }
+    permissions_code_parsed = xml_parsed.css('authorizationRequest > permissions > code').map(&:text)
+
+    expect(permissions).to eq permissions_code_parsed
+  end
+
+  it 'must have a redirect URL' do
+    redirect_url_parsed = xml_parsed.at_css('authorizationRequest > redirectURL').text
+    expect(redirect_url_parsed).to eq options[:redirect_url]
+  end
+
+  it 'must have a notification URL' do
+    notification_url = xml_parsed.at_css('authorizationRequest > notificationURL').text
+    expect(notification_url).to eq options[:notification_url]
+  end
+
+  context 'when an account is assigned' do
+    let(:account) { {} }
+    let(:options) { super().merge(account: account) }
+    let(:node) { super().at_css('> account') }
+
+    it 'can be an email' do
+      account[:email] = 'usuario@seusite.com.br'
+      expect( node.at_css('> email').text ).to eq account[:email]
```

Comment:
I recommend in tests like this to use:

```
it 'can be an email' do
       account[:email] = 'usuario@seusite.com.br'
       expect( node.at_css('> email').text ).to eq 'usuario@seusite.com.br'
end
```

Because `account` may change the e-mail or not :/

Be careful using code to test code :/